Content.Server/Disposal/Tube/Components/DisposalEntryComponent.cs

```diff
@@ -57,12 +45,9 @@ protected override Direction[] ConnectableDirections()
         /// </summary>
         public override Direction NextDirection(DisposalHolderComponent holder)
         {
-            if (holder.PreviousDirectionFrom != Direction.Invalid && holder.PreviousDirectionFrom == ConnectableDirections()[0])
+            if (holder.PreviousDirectionFrom != Direction.Invalid)
             {
-                var invalidDirections = new[] { ConnectableDirections()[0], Direction.Invalid };
-                var directions = Enum.GetValues(typeof(Direction))
-                    .Cast<Direction>().Except(invalidDirections).ToList();
-                return _random.Pick(directions);
+                return Direction.Invalid;
             }
```

Comment:
I don't quite get this change. Does PreviousDirectionFrom not get set anywhere? Because if it where to get set to anything other than Direction.Invalid NextDirection would just return Direction.Invalid.

Reply:
Also, PreviousDirectionFrom is as-of-yet-unrefactored code:
`public Direction PreviousDirectionFrom => (PreviousDirection == Direction.Invalid) ? Direction.Invalid : PreviousDirection.GetOpposite();`